### PR TITLE
Implement `.exclude`

### PR DESCRIPTION
**Pull request by EmilioCarrion**
Implemented the .exclude() method in the queryset. 
Implemented as a parametrization of the filter method where if we are excluding we invert the sql compose clause

### CODE REVIEW

File: orm/models.py [hunk 1]
```diff
@@ -94,14 +94,16 @@ def build_select_expression(self):
 
         return expr
 
-    def filter(self, **kwargs):
+    def filter(self, exclude=False, **kwargs):
```
Comment (tomchristie) [inline orm/models.py]: I wonder if we should use `_exclude`, to make it private-ish, and help ensure it shouldn't clash with any of the provided `kwargs`?

Comment (EmilioCarrion) [reply orm/models.py]: Totally true, I isolated the parsing logic in a private method avoiding any posible clash with field names

File: orm/models.py [hunk 2]
```diff
@@ -95,13 +95,21 @@ def build_select_expression(self):
         return expr
 
     def filter(self, **kwargs):
+        return self._parse_filters(**kwargs)
+
+    def exclude(self, **kwargs):
+        return self._parse_filters(exclude=True, **kwargs)
+
+    def _parse_filters(self, exclude=False, **kwargs):
```
Comment (rzane) [inline orm/models.py]: @EmilioCarrion, even with your latest changes, `exclude` could conflict one of the parameters used for filtering. Take a look at the following example:

```python
class Person(orm.Model):
    id = orm.Integer(primary_key=True)
    exclude = orm.String(max_length=100)

await Person.filter(exclude=True).all()
```

The column name `exclude` conflicts with the `exclude` parameter in `_parse_filters`.

One solution would be to change the definition of `_parse_filters` to the following:

```python
def _parse_filters(self, values, exclude=False):
```

Alternatively, you could rename `exclude` to `_exclude`.

File: orm/models.py [hunk 3]
```diff
@@ -95,14 +95,22 @@ def build_select_expression(self):
         return expr
 
     def filter(self, **kwargs):
+        return self._parse_filters(kwargs)
+
+    def exclude(self, **kwargs):
+        return self._parse_filters(kwargs, exclude=True)
+
+    def _parse_filters(self, filters, exclude=False):
```
Comment (rafalp) [inline orm/models.py:103]: Doesn't seem like this name describes what happens within the function. How about something like this:

```suggestion
    def _filter_query(self, filters, exclude=False):
```

File: orm/models.py [hunk 4]
```diff
@@ -154,7 +162,13 @@ def filter(self, **kwargs):
 
             clause = getattr(column, op_attr)(value)
             clause.modifiers['escape'] = '\\' if has_escaped_character else None
-            filter_clauses.append(clause)
+
+            clauses.append(clause)
+
+        if exclude:
+            filter_clauses.append(~sqlalchemy.sql.and_(*clauses))
```
Comment (rafalp) [inline orm/models.py:169]: Can we replace unary operator with `not_`? It will be more obvious that way.

**Review by aminalaee — APPROVED** [review]
@EmilioCarrion Thank you.

**Pull request merged**